benchmarks/cpu/CMakeLists.txt

```diff
@@ -0,0 +1,23 @@
+# TRACCC library, part of the ACTS project (R&D line)
+#
+# (c) 2024 CERN for the benefit of the ACTS project
+#
+# Mozilla Public License Version 2.0
+
+# Look for openMP, which is used for the CPU benchmark
+find_package(OpenMP)
+
+# Build the benchmark executable.
+traccc_add_executable(benchmark_cpu
+    "toy_detector_cpu.cpp"
+    LINK_LIBRARIES benchmark::benchmark benchmark::benchmark_main 
+    traccc::core traccc_benchmarks_common 
+    detray::core detray::utils vecmem::core)
+
+# Optimize the cpu benchmark (e.g. auto-vectorization)
+target_compile_options(traccc_benchmark_cpu PRIVATE
+    "-march=native" "-ftree-vectorize")
```

Comment:
No, we should not be doing this. :thinking:

Even if logically we'd like to set things up like this, we would need to explicitly check first whether the selected host compiler supports these flags. It might not.

But generally, such optimization flags are not meant to be specified by the project itself. But by the building/packaging system that builds the project.

In practice, you should set the `$CXXFLAGS` environment variable to the optimization flags that you'd like, and build the code like that. This way you could try out different optimization settings without changing the code at all.

Finally: I believe `-ftree-vectorize` is enabled at `-O3` by default. So just by building with `-DCMAKE_BUILD_TYPE=Release`, that flag should already be there. :thinking:

Reply:
OK...

> In practice, you should set the $CXXFLAGS environment variable to the optimization flags that you'd like, and build the code like that. This way you could try out different optimization settings without changing the code at all.

How can I do this exactly in CMakeListsts.txt? Or do you mean this should be set by the shell command

Reply:
OK I removed them now from the CMakeLists.txt